### ISSUE TITLE
https cannot show pics & messes up website

### ISSUE THREAD

**Issue by atulmehra** [opencart/opencart#9683]
**What version of OpenCart are you reporting this for?**
4.0.0.0_b
Git SHA - c67de320f639229b114e0d537acb5351c3709935

**Describe the bug**
When I access my site over https. it messes up display. With http, it works fine.

**To Reproduce**
Access website over https.
Setup your hosting serve to force all accesses to https. (I changed /opt/lampp/etc/extra/httpd-xampp.conf)

**Root cause:** The reason is as shown in the chrome dev tools
Mixed Content: The page at 'https://www.mysite.com/home' was loaded over HTTPS, but requested an insecure script '**http**://my-srerver-name.com/vaayu/catalog/view/javascript/jquery**/jquery-3.3.1.min.js'. This request has been blocked; the content must be served over HTTPS.**
**Expected behavior**
All images should be loaded with https.

**Screenshots / Screen recordings**

**Server / Test environment (please complete the following information):**
 - Local development? Deployed to a web server? Deployed on webserver. AWS ec2
 - Operating system - Ubuntu
 - PHP version - 8
 - Apache version
 - Browser(s) tested with Chrome Firefix
 - Have changed apache settings to force all http traffic to https
 - 

**Additional context**
I have hosted the website on ec2 which is behind a load balancer. SSL certificate is on load balancer. But that does not matter much.

**Comment by TheCartpenter** [comment]
.htaccess, config.php or admin/config.php file settings issue.

**Comment by atulmehra** [comment]
@TheCartpenter Thanks for reply. 
My limited understanding from code is that, in 4.0, config.php does not need to define HTTPS_SERVER, HTTPS_CATALOG etc.
If otherwise, could you show an example for 4.0?

**Comment by TheCartpenter** [comment]
You could re-test the protocol and path in a separate PHP file:





```if ((isset($_SERVER['HTTPS']) &amp;&amp; (($_SERVER['HTTPS'] == 'on') || ($_SERVER['HTTPS'] == '1'))) || $_SERVER['SERVER_PORT'] == 443) {	$protocol = 'https://';} elseif (!empty($_SERVER['HTTP_X_FORWARDED_PROTO']) &amp;&amp; $_SERVER['HTTP_X_FORWARDED_PROTO'] == 'https' || !empty($_SERVER['HTTP_X_FORWARDED_SSL']) &amp;&amp; $_SERVER['HTTP_X_FORWARDED_SSL'] == 'on') {	$protocol = 'https://';} else {	$protocol = 'http://';}```

```$protocol . $_SERVER['HTTP_HOST'] . rtrim(rtrim(dirname($_SERVER['SCRIPT_NAME']), 'install'), '/.\\') . '/');```

under HTTPS and see the results.

**Comment by mhcwebdesign** [comment]
Please be aware that the master branch is not a production-ready release, it is a development in progress.
You are welcome to test it, but then please do it for the latest one. Quite obviously you were using an earlier version of the 4.x master branch, because this bug does not exist anymore.

The header.twig contains this:
`<script src="{{ jquery }}" type="text/javascript"></script>`

The **jquery** variable is set in the header.php controller:
`$data['jquery'] = 'catalog/view/javascript/jquery/jquery-3.6.0.min.js';`

The resulting output is this:
`<script src="catalog/view/javascript/jquery/jquery-3.6.0.min.js" type="text/javascript"></script>`

And it is based on this in the header.twig:
`<base href="{{ base }}"/>`

Where base is set by the header.php controller as follows:
`$data['base'] = $this->config->get('config_url');`

The **config_url** is set in the startup/setting.php as follows:
`$this->config->set('config_url', HTTP_SERVER);`

And the HTTP_SERVER is defined in the config.php, and it should of course start with 'https://'!

In summary, there is nothing wrong here on this with the current master branch code, so nothing to be fixed here!

**Comment by atulmehra** [comment]
Awesome. Thnaks @mhcwebdesign . Let me try latest one. btw, are monthly or stable 4.0 releases tagged? So we can periodically upgrade the tagged ones.  Our release is also in beta, so we would prefer to be beta users of OC & get involved.
